### ISSUE TITLE
Connection info

### ISSUE THREAD

**Issue by arctus** [Schack17/PaulFocusNovus#1]
Hi,

Not sure whether this is the most appropriate way to contact you. Apparently yours code is only one of couple that is available for the Paul Novus. I would really love to add Novus to my openhab system to control the fans (and only the fans).

The questions that I have:
1. Do you mind to tell me how do you connect this to your RS485 adapter, which terminals do you use (any photo would help)?
2. As I understand from the code, you emulate a touch panel, is this right?

**Comment by taliesin** [comment]
Hi,
   as you haven't been answered by the owner of the repo yet, I'll try instead ...
Note: The whole discussion about RS485 for Paul Novus is in German, here are a few links:

- https://knx-user-forum.de/forum/öffentlicher-bereich/knx-eib-forum/diy-do-it-yourself/997148-steuereung-lüfterstufe-kwl-paul-focus-200-novus-300
- https://www.haustechnikdialog.de/Forum/t/153632/Paul-Novus-und-der-RS485-Anschluss

Taken from them:
1.) pinning

Contact * color * Signal * RJ45
X1.1 * red * 24V * Pin 8
X1.2 * white * T/R+ * Pin 5
X1.3 * yellow * T/R- * Pin 4
X1.4 * black * GND * Pin 1,3,2
X1.5 * / * shield * /

2.) This repo does not allow 'write access' to the master controller, it only reads the information passed between a connected touch panel and the master.

As of my knowledge there is currently no direct way to drive the Paul Novus 300 via RS485, except a rather weird method using a relay, see https://github.com/arduinoProjz/paulFanSpeed
Major trouble seems to be the tight timing requirements demanded by the main controller.

The repo owner (as of July 2019) uses the 0-10V input (connector X10.9 and X10.10) to change the ventilation speed.

From my point of view there should be a way to use only the RS485 bus to manage both 'reading' and 'writing'. However there would still have to be 2 RS485 buses, one for the main controller and one for touch panel and some embedded gadget in the middle, which emulates the main controller for the touch panel and the touch panel for the main controller. This way it should be possible to decouple (and modify) data to and from the touch panel.

This should allow for leaving away the touch panel if not needed, later on.

**Comment by arctus** [comment]
Thank you for the detailed answer.

I did find those forums, but as not a native speaker, those are hard to understand.

Naturally, I do not only want to write (0-10V), but also read the data so that I could show that in the home automation.

I have tried the paulFanSpeed, but I do have problems with it - I receive garbage on the rs485 (maybe  I do something wrong with 9bit uart, BTW, can you share with me a full-proof way to get 9bit working on arduino (or esp32 better)?)

Agree, that the best way is to have it as intrusion free as possible, I even thought about buying a LED panel set arduino to read out LEDs and short buttons when needed.

**Comment by Schack17** [comment]
@arctus Sorry for my late answer. I hate the github notifications... They are not working good.
I have a Novus 300 and I use a standard RS-485-USB-Stick with a raspberry pi. I think my Novus 300 works with normal 8 Bit and other Novus models (e.g. 450) work not with 8 bit or other protocols.
With my Novus 300 I can read out the 4 temperatures: Outdoor - House in after defroster, Incoming air - rooms in, Indoor / Outgoing air - rooms out and Outgoing air - house out.
You can see it in the documentation: https://github.com/Schack17/PaulFocusNovus/wiki/RS485-Bus-Paul-Novus-300-(Focus-200)
And the very dirty script: https://github.com/Schack17/PaulFocusNovus/blob/master/paulRS485Logger.py
I can't write on the bus. To manage the fan speed I use a 0-10V output from tinkerforge.
At the moment I am porting my house automation from openhab to home assistant because mqtt is working much better. And my python knowhow is better than java.
The tinkerforge 0-10V-Script is now working with home assistant.
And the next task is porting the RS485 script.

**Comment by arctus** [comment]
@Schack17 Thank you for a reply. 
if you control 0-10V, that means you are not using the TFT screen anymore, right? or they can both function simultaneously? 
Regarding openhab, I am pretty happy with openhab now that I use Jython (more or less python 2.X) instead of that strange Java-like language for writing rules. Also, I migrated openhab to docker running on a Ubuntu VM on Proxmox in Proliant Server. I have found this setup to be the most efficient.

**Comment by Schack17** [comment]
@arctus Yes. The TFT has to use the mode "automatic" or "sensor". Only in this mode the 0-10V-Input is used.
I have very much trouble with openhab and mqtt. Openhab send very much to the mqtt and it dies. Home assistant is working very good with mqtt and discover every mqtt device automaticly and allows to config it over the gui.

**Comment by andrejs2** [comment]
Hi,
@Schack17 , just saw your page on access to paul novus. Did you manage to port to HA? I assume communication is via mqtt? Do you have any interest to share your project? 

I'm very much interested in getting Paul Novus on HA and get some sensor readings and perhaps control the fans speed. Anyway, thanks for your efforts!!

**Comment by Schack17** [comment]
Hi @andrejs2,
my novus300-"custom_component" has a very dirty code. 🤦‍♂️
And it is a mix to read the temperatures over the RS-485-Bus and write the 0-10V output via tinkerforge brick.
I have updated my repo with the only RS485 part:
https://github.com/Schack17/PaulFocusNovus/tree/master/home-assistant/custom_components/novus300bus
With this you can read the temperatures from the bus. The sensors are currently hard coded.

Have you seen this project:
https://github.com/oversc0re/paul_novus_hass
oversc0re has reverse engineered the bypass state and the time remaining to filter exchange - and the fan speed?
Very interesting.

Do you know the [default modbus module from home assistant](https://www.home-assistant.io/integrations/modbus/).
For some time now I've been wondering whether the Novus connection could also work with it?

**Comment by andrejs2** [comment]
@Schack17 Thank you very much for your quick reply! 

I'm sorry I'm not so keen in this stuff and not an engineer at all...I can't help, but I'm willing to test :) 

Anyway I'll try connecting first and will see how it's working. And indeed I saw oversc0re's project, thank you. 
I'll follow you guys, if you find out some more interesting stuff!
Again, thanks for all your efforts and energy you put into this!
